scripts/cleanup_edomain_templates.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python2
+
+"""This script takes an provider and edomain as optional parameters, and
+searches for old templates on specified provider's export domain and deletes
+them. In case of no --provider parameter specified then this script
+traverse all the rhevm providers in cfme_data.
+"""
+
+import argparse
+import datetime
+from threading import Lock, Thread
+
+from ovirtsdk.api import API
+from utils.conf import cfme_data
+from utils.conf import credentials
+from utils.ssh import SSHClient
+from utils.log import logger
+from utils.wait import wait_for
+
+lock = Lock()
+
+
+def parse_cmd_line():
+    parser = argparse.ArgumentParser(argument_default=None)
+    parser.add_argument("--edomain", dest="edomain",
+                        help="Export domain for the remplate", default=None)
+    parser.add_argument("--provider", dest="provider",
+                        help="Rhevm provider (to look for in cfme_data)",
+                        default=None)
+    args = parser.parse_args()
+    return args
+
+
+def make_ssh_client(rhevip, sshname, sshpass):
+    connect_kwargs = {
+        'username': sshname,
+        'password': sshpass,
+        'hostname': rhevip
+    }
+    return SSHClient(**connect_kwargs)
+
+
+# get the domain edomain path on the rhevm
+def get_edomain_path(api, edomain):
+    edomain_id = api.storagedomains.get(edomain).get_id()
+    edomain_conn = api.storagedomains.get(edomain).storageconnections.list()[0]
+    return (edomain_conn.get_path() + '/' + edomain_id,
+            edomain_conn.get_address())
+
+
+def cleanup_empty_dir_on_edomain(path, edomainip, sshname, sshpass):
```

Comment:
The path is fetched much before any operation is performed on the edomain. line# 206. Assuming the provider edomain is accessible to perform api operation. The OVF error is caused, when template is deleted but not the directory. This function is called in finally block, to make sure there are no empty directories left on edomain.